mongo/single_result.go

```diff
@@ -28,6 +29,22 @@ type SingleResult struct {
 	reg *bsoncodec.Registry
 }
 
+// NewSingleResultFromBytes creates a SingleResult with an underlying Cursor pre-loaded with the provided contents.
+func NewSingleResultFromBytes(contents []byte) *SingleResult {
```

Comment:
We also need a way to allow the `SingleResult` to return errors. Add a 2nd `error` argument or consider accepting an `io.Reader` instead of a `[]byte, error` (note that would probably require refactoring how the input is used to create a `Cursor`, but it's a more standardized way to provide bytes+errors to a function, so worth consideration).

Reply:
Changed `NewMockSingleResult` and `NewMockCursor` to take in an `error` and a `bsoncodec.Registry`.